[h4ow3i] iP

h4ow3i opened the pull request:
**_Edgar_**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))


Edgar frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/ljy0422/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

h4ow3i committed:
level 0

h4ow3i committed:
level 1

h4ow3i committed:
level 2

h4ow3i committed:
Level 3

h4ow3i committed:
Level 4

h4ow3i committed:
Level 5

h4ow3i committed:
Level 6

h4ow3i committed:
A-TextUiTesting

h4ow3i committed:
level 7

h4ow3i committed:
level 8 (handle date and time inputs)

h4ow3i committed:
add A-MoreOOP

h4ow3i committed:
Add A-Packages

h4ow3i committed:
Merge commit 'f837ddb7d2b77f6654758f93d056420a580787ca'

h4ow3i committed:
Add JUnit

h4ow3i committed:
delete

h4ow3i committed:
A-JAR

h4ow3i committed:
Add javadoc to all classes

h4ow3i committed:
Add Coding Standard

ChuaZenKhoon commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,180 @@
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    public Command firstParse(String input) {
```
I understand this first parses the input and gets the command, maybe parseCommand would be clearer 😄

ChuaZenKhoon commented on src/main/java/EventCommand.java:
```diff
@@ -0,0 +1,20 @@
+    private LocalDateTime end;
+
+    public EventCommand(String event, LocalDateTime start, LocalDateTime end) {
+        this.event = event;
```
Nice use of start and end for event and due for deadline to differentiate the usage! 👍

ChuaZenKhoon commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.Scanner;
+
+public class Storage {
+    private static final String FILE_PATH = "./data/duke/.txt";
```
Nice use of Uppercase all for static final variables! 👍

ChuaZenKhoon reviewed (commented):
Hi, I'm the 2nd reviewer so I'll focus on just naming quality. I used this commit as the latest one(JAR) doesn't have much to go on off.
Overall really nice!

h4ow3i committed:
Add find feature

h4ow3i committed:
Merge branch 'branch-A-CodingStandard'

h4ow3i committed:
Merge branch 'branch-Level-9'

h4ow3i committed:
Add checkstyle

h4ow3i committed:
edit build.gradle for checkstyle

h4ow3i committed:
testing

h4ow3i committed:
testing

h4ow3i committed:
Add level 10(GUI)

h4ow3i committed:
Add assertions to codebase

The codebase lacks assertions, leaving potential vulnerabilities due to unexpected null values or exceeding maximum capacities.

 Adding assertions enhances the robustness of the code by validating inputs and ensuring that the program behaves as expected under various conditions.

 Implement assertions throughout the codebase to validate inputs, ensuring that critical conditions are met before proceeding with operations.

 Assertions provide a proactive approach to identify and handle potential issues, enhancing code reliability and reducing the risk of runtime errors.

{Any other relevant info} - By incorporating assertions, we improve code quality and facilitate debugging processes, contributing to a more stable and resilient software system.

h4ow3i committed:
Merge pull request #1 from h4ow3i/branch-A-Assertions

Add assertions to codebase

laney0808 commented on src/main/java/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public class DeadlineCommand extends Command {
+
+    private String taskDescription;
```
Clear and comprehensible variable names; Appropriate accessibility for class attributes.

laney0808 commented on src/main/java/edgar/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+    private ImageView displayPicture;
+
+    private DialogBox(String text, Image img) {
+        try {
```
Error properly handled, with spacing and indentation conforming to coding standards

laney0808 reviewed (commented):
Generally conforming coding standards. Code is clear and understandable. There is use of non-specific variable name such as 'tmp', but does not hinder understanding of the code.

h4ow3i committed:
Fixed some bugs and added auto exit function to chatbot

h4ow3i committed:
Add User Guide

h4ow3i committed:
Fixed some typos

h4ow3i committed:
Fix GUI window sizing

h4ow3i committed:
Fix bugs in GUI
